app/src/main/java/de/tap/easy_xkcd/database/RealmComic.java

```diff
@@ -286,17 +281,18 @@ public static Bitmap getOfflineBitmap(int comicNumber, Context context, PrefHelp
         }
 
         Bitmap mBitmap = null;
+        String comicFileName = comicNumber + ".png";
         try {
             File sdCard = prefHelper.getOfflinePath();
             File dir = new File(sdCard.getAbsolutePath() + OFFLINE_PATH);
-            File file = new File(dir, String.valueOf(comicNumber) + ".png");
+            File file = new File(dir, comicFileName);
             FileInputStream fis = new FileInputStream(file);
             mBitmap = BitmapFactory.decodeStream(fis);
             fis.close();
         } catch (IOException e) {
             Timber.e( "Image not found, looking in internal storage");
             try {
-                FileInputStream fis = context.openFileInput(String.valueOf(comicNumber));
+                FileInputStream fis = context.openFileInput(comicFileName);
```

Comment:
For legacy reasons, when the files are saved to interal storage, the file extension is actually emitted, so the parameter should still `String.valueOf(comicNumber)` to maintain backwards compatibility. I already fixed this in my latest commit.